Modules/System/Cryptography Management/src/CertificateRequestImpl.Codeunit.al

```diff
@@ -0,0 +1,153 @@
+            HashAlgorithm::SHA384:
+                DotNetHashAlgorithmName := DotNetHashAlgorithmName.SHA384();
+            HashAlgorithm::SHA512:
+                DotNetHashAlgorithmName := DotNetHashAlgorithmName.SHA512();
+        end;
+
+        DotNetCertificateRequest := DotNetCertificateRequest.CertificateRequest(SubjectName, DotNetRSACryptoServiceProvider, DotNetHashAlgorithmName, DotNetRSASignaturePadding);
+    end;
+
+    procedure AddX509BasicConstraintToCertificateRequest(CertificateAuthority: Boolean; HasPathLengthConstraint: Boolean; PathLengthConstraint: Integer; Critical: Boolean)
+    var
+        DotNetX509BasicConstraintsExtension: DotNet X509BasicConstraintsExtension;
+    begin
+        DotNetCertificateRequest.CertificateExtensions.Add(
+            DotNetX509BasicConstraintsExtension.X509BasicConstraintsExtension(
+                CertificateAuthority, HasPathLengthConstraint, PathLengthConstraint, Critical));
+    end;
+
+    procedure AddX509EnhancedKeyUsageToCertificateRequest(OidValues: List of [Text]; Critical: Boolean)
+    var
+        DotNetX509EnhancedKeyUsageExtension: DotNet X509EnhancedKeyUsageExtension;
+        DotNetOidCollection: DotNet OidCollection;
+        DotNetOid: DotNet Oid;
+        Item: Text;
+    begin
+        DotNetOidCollection := DotNetOidCollection.OidCollection();
+        foreach Item in OidValues do
+            DotNetOidCollection.Add(DotNetOid.Oid(Item));
+
+        DotNetCertificateRequest.CertificateExtensions.Add(
+            DotNetX509EnhancedKeyUsageExtension.X509EnhancedKeyUsageExtension(DotNetOidCollection, Critical));
+    end;
+
+    procedure AddX509KeyUsageToCertificateRequest(X509KeyUsageFlags: Integer; Critical: Boolean)
+    var
+        DotNetX509KeyUsageExtension: DotNet X509KeyUsageExtension;
+        DotNetX509KeyUsageFlags: Dotnet X509KeyUsageFlags;
+    begin
+        DotNetX509KeyUsageFlags := X509KeyUsageFlags;
+        DotNetCertificateRequest.CertificateExtensions.Add(
+            DotNetX509KeyUsageExtension.X509KeyUsageExtension(DotNetX509KeyUsageFlags, Critical));
+    end;
+
+    procedure CreateSigningRequest(var SigningRequestPemString: Text)
+    var
+        Base64Convert: Codeunit "Base64 Convert";
+        TempBlob: Codeunit "Temp Blob";
+        SigningRequestOutStream: OutStream;
+        SigningRequestInStream: InStream;
+        Pem: TextBuilder;
+        B64: Text;
+        Pos: Integer;
+    begin
+        TempBlob.CreateOutStream(SigningRequestOutStream);
+        TempBlob.CreateInStream(SigningRequestInStream);
+        CreateSigningRequest(SigningRequestOutStream);
+        B64 := Base64Convert.ToBase64(SigningRequestInStream);
+
+        Pos := 1;
+        Pem.AppendLine('-----BEGIN CERTIFICATE REQUEST-----');
+        while Pos < StrLen(B64) do begin
+            Pem.AppendLine(CopyStr(B64, Pos, 64));
+            Pos += 64;
+        end;
+        Pem.Append('-----END CERTIFICATE REQUEST-----');
```

Comment:
TextConstant please. Don't forget to lock it.

Reply:
Fixed.